smtp-server/src/lib.rs

```diff
@@ -656,7 +691,15 @@ where
                                 // long, non-RFC5322-compliant, etc.).
                                 // Couldn't find the RFC reference
                                 // anywhere, though.
-                                simple_handler!(decision);
+                                let mut n_decisions = 0;
+                                while let Some(decision) = decision_stream.next().await {
+                                    n_decisions += 1;
+                                    if n_decisions > expected_n_decisions {
+                                        break;
```

Comment:
```suggestion
                                        panic!("got more decisions in handle_mail return than the expected {}", expected_n_decisions);
```

Otherwise the error message could underestimate the number of decisions given